Mobl_ios_Callisto-master/MealTrackingPilot/LogCommonMeal/Views/OccasionPicker.swift

```diff
@@ -0,0 +1,372 @@
+        selectionImageView.centerXAnchor.constraint(equalTo: self.centerXAnchor).isActive = true
+        selectionImageView.centerYAnchor.constraint(equalTo: self.centerYAnchor).isActive = true
+    }
+    
+    private func setupSelectionIndicator() {
+        selectionIndicator.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(selectionIndicator)
+        
+        selectionIndicator.widthAnchor.constraint(equalToConstant: 2.0).isActive = true
+        selectionIndicator.heightAnchor.constraint(equalTo: self.heightAnchor).isActive = true
+        selectionIndicator.trailingAnchor.constraint(equalTo: self.centerXAnchor, constant:itemSpan/2).isActive = true
+        selectionIndicator.centerYAnchor.constraint(equalTo: self.centerYAnchor).isActive = true
+    }
+    
+    // MARK: Infinite Scrolling Helpers
+    
+    private func generateInfinityItemsMultiplier() -> Int {
+        if scrollingStyle == .default {
+            return 1
+        }
+        
+        if numberOfItemsByDataSource > 100 {
+            return 100
+        } else if numberOfItemsByDataSource < 100 && numberOfItemsByDataSource > 50 {
+            return 200
+        } else if numberOfItemsByDataSource < 50 && numberOfItemsByDataSource > 25 {
+            return 400
+        } else {
+            return 800
+        }
+    }
+    
+    // MARK: UI handlers
+    
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+        
+        if !setupHasBeenDone {
+            configureSetup()
+            setupHasBeenDone = true
+        }
+    }
+    
+    func indexForItem(_ item: Int) -> Int {
+        return item % (numberOfItemsByDataSource > 0 ? numberOfItemsByDataSource : 1)
+    }
+    
+    // MARK: - Actions
+    
+    /**
+     Selects the nearby to middle item that matches with the provided index.
+     
+     - parameter item: A valid index provided by Data Source.
+     */
+    private func selectedNearbyToMiddleItem(_ item: Int) {
+        currentSelectedItem = item % numberOfItemsByDataSource
+        collectionView.reloadData()
+        
+        if numberOfItemsByDataSource > 0 && collectionView.numberOfItems(inSection: 0) > 0 {
+            let indexOfSelectedItem = visibleIndexOfSelectedItem()
+            setContentOffset(CGFloat(indexOfSelectedItem) * itemSpan - endCapSpan, animated: false)
+            
+            delegate?.occasionPicker?(self, didSelectItem: currentSelectedItem, index: currentSelectedIndex)
+        }
+    }
+    
+    private func setContentOffset(_ offset: CGFloat, animated: Bool) {
+        var offsetPoint = CGPoint.zero
+        switch scrollingDirection {
+        case .horizontal:
+            offsetPoint.x = offset
+        case .vertical:
+            offsetPoint.y = offset
+        }
+        
+        if animated {
+            isAnimating = true
+        }
+        
+        collectionView.setContentOffset(offsetPoint, animated: animated)
+    }
+    
+    var itemSpan: CGFloat {
```

Comment:
Migrate this span properties  before to // MARK: Initialization

Reply:
Done